src/components/ui/Tag/Tag.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import ResizableComponent from '../ResizableComponent/ResizableComponent';
+import classNames from 'classnames';
+
+interface TagProps {
+  id: string;
+  label: string;
+  onDelete?: (id: string) => void;
+}
+
+const Tag = ({ id, label, onDelete }: TagProps) => {
+  const [text, setText] = useState(label);
+  const [deleteMode, setDeleteMode] = useState(false);
+
+  useEffect(() => {
+    deleteMode ? setText('Delete?') : setText(label);
+  }, [deleteMode]);
```

Comment:
During my testing, I noticed that the component didn't update when the label prop was changed.

I suspect that's due to the internal `text` state, which isn't necessarily bad. But we could improve it a bit by eliminating the text state and the useEffect. It could be even cleaner!

We can do this by doing something similar to the code below:

```tsx

const [deleteMode, setDeleteMode] = useState(false);
const text = !deleteMode ? label : "Delete?";

```